### ISSUE TITLE
RFE: auto-populate graph from function name & args

### ISSUE THREAD

**Issue by ankostis** [cedricleroy/pyungo#35]
I propose to auto-populate `Graph.add_node()` calls based on function & argument-names (using python's [`inspect`](https://docs.python.org/3/library/inspect.html) standard-library's module), and allow some form of string-filtering on the function/args names.
Knowingly this would work for (singular) outputs only. 

The proposal is easier to explain with sample client code:

```python
def funcname_chopper(funcname):
    for prefix in ['calc_', 'compute_', 'make_']:
        if funcname.startswith(prefix):
            return  prefix[len(prefix):]

graph = pyungo.Graph(
    outname_converter=funcname_chopper)

# equivalent to: register(inputs=['a', 'b'], outputs=['c']
@graph.register
def make_c(a, b):
    return a+b


# equivalent to: register(inputs=['a', 'b', 'c'], outputs=['make_d']
@graph.register(inpname_converter=lambda n: n[5:], outname_converter=None)
def calc_d(some_a, look_b, stop_c):
    return a+b

```

**Comment by ankostis** [comment]
My mistake, this shortcut enhancement is best fit for the `register` annotation, not for the `add_node()`.

**Comment by cedricleroy** [comment]
I thought of some uses from the `inspect` module, but more on the checking side of things (e.g. enforce inputs names with function argument names). I am not sure I am a big fan of defining the output name in the function name though, but not totally against it. Another option would be to use type hints for defining the output name (with `__annotations__`), but this would inhibit the type hint feature:
```python
@graph.register
def make_c(a, b) -> 'c':
    return a + b
```

**Comment by cedricleroy** [comment]
Should be closed by #38: 

```python
# equivalent to: register(inputs=['a', 'b'], outputs=['c'])
@graph.register()
def make_c(a, b):
    c = a + b
    return c
```
